common/djangoapps/tedix_ro/views.py

```diff
@@ -197,8 +199,16 @@ def manage_courses(request):
                         send_mail(subject, txt_message, from_address, [parent.user.email], html_message=html_message)
 
                 if form.cleaned_data['send_sms']:
-                    # sending sms logic to be here
-                    pass
+                    parent = student.parents.first()
+                    context.update({
+                        'student_name': student.user.profile.name or student.user.username,
+                        'courses_due_dates_url': urljoin(settings.LMS_ROOT_URL, reverse('personal_due_dates')),
+                    })
+                    sms_message = render_to_string(
+                        'sms/manage_course_notify.txt',
+                        context
+                    )
+                    sms_client.send_message(parent.phone, sms_message)
```

Comment:
I have no insight of this project but isn't it better to do any network interaction in background?
Based on this code Instructor doing some global action to send sms's to multiple parents.
So theoretically it can be a list of parent and it can cause page response delay for a User(instructor).

Reply:
Noticed, will be changed in one of the next releases.